src/pulse3D/excel_writer.py

```diff
@@ -221,6 +225,56 @@ def write_xlsx(
     interpolated_data_period = (
         w[INTERPOLATION_VALUE_UUID] if plate_recording.is_optical_recording else INTERPOLATED_DATA_PERIOD_US
     )
+
+    # get stim meta data
+    if include_stim_protocols:
+        protocols_id_list = []
+        unassigned_wells = []
+        stim_protocols_dict = {
+            "Title": {
+                "Unassigned Wells": "Unassigned Wells:",
+                "title_break": "",
+                "Protocol ID": "Protocol ID:",
+                "Stimulation Type": "Stimulation Type:",
+                "Run Until Stopped": "Run Until Stopped:",
+                "Wells": "Wells:",
+                "Subprotocols": "Subprotocols:",
+            }
+        }
+        for well in plate_recording.wells:
+            if (well_data := json.loads(well[STIMULATION_PROTOCOL_UUID])) is not None:
+                protocol_id = well_data.get("protocol_id")
+                well_id = well[WELL_NAME_UUID]
+                if protocol_id not in protocols_id_list:
```

Comment:
It looks like you're just using this list to keep track of whether or not a given protocol_id is in `stim_protocols_dict` yet, so you could just check the dict directly instead: `if protocol_id in stim_protocols_dict:`